### PR TITLE
check timing after each iteration

### PR DESCRIPTION
**Pull request by patripfr**
This changes make sure we check after each ceres iteration if the optimization time is still below `ba_max_time_seconds`. Previously it would always perform `num_iters`, also if already 1 iteration would exceed the time limit.

### CODE REVIEW

File: algorithms/map-optimization/src/outlier-rejection-solver.cc [hunk 1]
```diff
@@ -137,11 +137,13 @@ void findOutlierLandmarks(
...
     const ceres::Solver::Options& solver_options, int num_iters,
     OptimizationProblem* optimization_problem,
-    OutlierRejectionCallback* callback, OptimizationProblemResult* result) {
+    OutlierRejectionCallback* callback, OptimizationProblemResult* result,
+    const double* max_solver_time_s = nullptr) {
   CHECK_NOTNULL(optimization_problem);
   CHECK_NOTNULL(callback);
   // 'result' can be a nullptr.
 
+  timing::TimerImpl timer_solve_inner("BA: Solve Interal");
```
Comment (LBern) [inline algorithms/map-optimization/src/outlier-rejection-solver.cc:146]: does it start the timer here? I'm wondering about the first iteration of the loop?

File: algorithms/map-optimization/src/outlier-rejection-solver.cc [hunk 2]
```diff
@@ -157,17 +159,33 @@ ceres::TerminationType solveStep(
...
-
-  ceres::Solver::Summary summary;
-  ceres::Solve(local_options, &problem, &summary);
-
-  // Save the solver summary of each iteration.
-  if (result != nullptr) {
-    result->solver_summaries.emplace_back(summary);
+  if (max_solver_time_s) {
+    double remaining_time = *max_solver_time_s;
+    local_options.max_num_iterations = 1;
```
Comment (LBern) [inline algorithms/map-optimization/src/outlier-rejection-solver.cc:165]: have you noticed any difference in doing a manual loop around ceres in comparison to doing 3 inner loops? I assume the problem is kept allocated

**Review by LBern — APPROVED** [review]
looks good, thansk. Only two questions from my side.

File: algorithms/map-optimization/src/outlier-rejection-solver.cc (hunk 1 above)
Comment (patripfr) [reply algorithms/map-optimization/src/outlier-rejection-solver.cc:146]: yes it does, see:
https://github.com/ethz-asl/aslam_cv2/blob/610b70a753a589bba215670b67fd1c75fa7fc01e/aslam_cv_common/src/timer.cc#L61

**Review by LBern — APPROVED** [review]
very nice

**Pull request merged**